Integrations/integration-cisco-ise.yml

```diff
@@ -0,0 +1,543 @@
+description: Next-generation secure network access
+configuration:
+- display: Cisco ISE server URL. For example, https://123.123.123.65
+  name: serverURL
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: Server port. For example, 9060
+  name: serverPort
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: Cisco ISE username
+  name: credentials
+  defaultvalue: ""
+  type: 9
+  required: true
+- display: Use system proxy settings
+  name: useProxy
+  defaultvalue: "true"
+  type: 8
+  required: false
+- display: Trust any certificate (unsecure)
+  name: insecure
+  defaultvalue: "false"
+  type: 8
+  required: false
+script:
+  script: |-
+    ''' IMPORTS '''
+    import requests
+    import json
+    import os
+    import sys
+    import re
+    from requests.auth import HTTPBasicAuth
+    from requests.packages.urllib3.exceptions import InsecureRequestWarning
+
+    # disable insecure warnings
+    requests.packages.urllib3.disable_warnings()
+
+    ''' GLOBAL VARS '''
+
+    BASE_URL=re.sub("/[\/]+$/", "", demisto.params().get('serverURL'))
+    SERVER_PORT=demisto.params().get('serverPort')
+    SERVER_URL=BASE_URL + ':' + str(SERVER_PORT)
+
+    USERNAME=demisto.params().get('credentials').get('identifier')
+    PASSWORD=demisto.params().get('credentials').get('password')
+
+    USE_SSL = not demisto.params().get('insecure', False)
+
+    ISE=requests.session()
+    ISE.auth = (USERNAME, PASSWORD)
+    ISE.verify = USE_SSL
+    ISE.disable_warnings = True
+    ISE.timeout = 5
+    ISE.headers.update({'Accept':'application/json', 'Connection': 'keep_alive'})
+
+    ''' HELPER FUNCTIONS '''
+
+    def load_proxy():
+        """ Loads the system configured proxy if enabled in configuration """
+        proxy = {}
+        if "proxy" in demisto.params():
+            proxy["http"] = os.environ["http_proxy"]
+            proxy["https"] = os.environ["https_proxy"]
+        return proxy
+
+    PROXY = load_proxy()
+
+    def is_mac(mac):
+        """
+        Test for valid mac address
+        :param mac: MAC address in the form of AA:BB:CC:00:11:22
+        :return: True/False
+        """
+
+        if re.search(r'([0-9A-F]{2}[:]){5}([0-9A-F]){2}', mac.upper()) is not None:
+            return True
+        else:
+            return False
+
+    def http_request(method, url_suffix, params_dict, data=None, headers=None):
+
+        url = SERVER_URL + url_suffix
+        LOG('running %s request with url=%s' % (method, url))
+
+        try:
+            if method == 'GET':
+                if headers is not None:
+                    ISE.headers.update(headers)
+                result = ISE.get(url, auth=HTTPBasicAuth(USERNAME, PASSWORD), verify=USE_SSL)
+
+                if result.status_code == 200:
+                    return result.text
+                elif result.status_code == 404:
```

Comment:
Handle `404` status code better: in `reauthenticate-endpoint` and in `get-endpoint-details`